### PR TITLE
[Scala] Properly handle csv collectionFormat

### PR DESCRIPTION
**Pull request by gmarz**
### PR checklist

- [x] Read the [contribution guidelines](https://github.com/swagger-api/swagger-codegen/blob/master/CONTRIBUTING.md).
- [x] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh` and `./bin/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`.
- [x] Filed the PR against the correct branch: master for non-breaking changes and `3.0.0` branch for breaking (non-backward compatible) changes.

### Description of the PR

Adds support for `"collectionFormat": "csv"`, which previously use to cause an exception.

So now

```json
{
  "name": "foo",
  "in": "path",
  "type": "array",
  "items": {
    "type": "string"
  },
  "collectionFormat": "csv"
}
```

will result in

`def someOperation(foo: List[String])`

(Oops...another fix snuck in here -- defaulting `None` for case classes...I can open a separate PR for this if desired)

### CODE REVIEW

**Comment by wing328** [comment]
@gmarz thank for the PR.

cc @clasnake @foxmk @ramzimaalej

**Comment by wing328** [comment]
> (Oops...another fix snuck in here -- defaulting None for case classes...I can open a separate PR for this if desired)

Please file a separate PR for defaulting None  in case classes as I found other Scala SDK not doing that so I think it's subject to more discussion before we can add that to the Scala generator.

**Comment by gmarz** [comment]
@wing328 sure thing, I'll try and get to this soon.

**Comment by gmarz** [comment]
> Please file a separate PR for defaulting None in case classes 

@wing328 I've reverted the change from this PR.

Perhaps it's appropriate to add it to #6538 since it's related. Let me know, otherwise I'll open another PR.

**Comment by wing328** [comment]
> Perhaps it's appropriate to add it to #6538 since it's related. Let me know, otherwise I'll open another PR.

I would suggest filing another PR for the change.

Btw, what issues did you run into if not defaulting to None? A spec to reproduce the issue would be helpful in reviewing the change.

**Comment by gmarz** [comment]
> Btw, what issues did you run into if not defaulting to None? A spec to reproduce the issue would be helpful in reviewing the change.

It's not really an issue per se, it's more for convenience. I'll open a PR with an example so we can keep the discussion here relevant to the change in this PR.

**Comment by wing328** [comment]
I also left some comments in https://github.com/swagger-api/swagger-codegen/pull/6538. Please take a look when you've time.

File: modules/swagger-codegen/src/main/resources/scala/apiInvoker.mustache [hunk 1]
```diff
@@ -52,6 +52,9 @@ class ApiInvoker(val mapper: ObjectMapper = ScalaJsonUtil.getJsonMapper,
   def escape(value: String): String = {
     URLEncoder.encode(value, "utf-8").replaceAll("\\+", "%20")
   }
+  def escape(values: List[String]): String = {
```
Comment (wing328) [inline modules/swagger-codegen/src/main/resources/scala/apiInvoker.mustache]: Later we will need to add another method `def escape(values: List[String], format: String): String = { ... }` to support other collection formats.

cc @clasnake

**Comment by gmarz** [comment]
@wing328 

> Please file a separate PR for defaulting None in case classes as I found other Scala SDK not doing that so I think it's subject to more discussion before we can add that to the Scala generator.

Done. https://github.com/swagger-api/swagger-codegen/pull/6790

**Pull request merged**